Feature/509 create basic challenge form

deuvede24 opened the pull request:
### **Description:**  
- Created a standalone form for mentors to create challenges with basic validations (`onSubmit()` method).  
- Added `createChallenge()` method in `ChallengeService` using mocked data. The real endpoint can be used by uncommenting the code.  
- Updated `core-routing.module.ts` with the `challenges/create` route.  
- Added `CreateChallenge` interface for form structure.

### **Next Steps:**  
- Connect the form to the backend once the backend PR is approved.  
- Move the route to the mentor panel when it’s ready.

deuvede24 committed:
feat: add create challenge method in challenge.service.ts

deuvede24 committed:
feat: convert challenge-form into a standalone component
& chore: comment out Cypress auto-generated test file

deuvede24 committed:
fix: adjust route order in core-routing to resolve conflicts

deuvede24 committed:
feat: add 'CreateChallenge' interface for challenge creation

deuvede24 committed:
feat: add mock implementation for createChallenge in service

deuvede24 committed:
refactor: update onSubmit method with comments and remove unused code
Comments to the `onSubmit` method to document future changes (mentor-specific path and guards). Removed commented-out code for better readability.

deuvede24 committed:
chore: remove unused Cypress test file for standalone component"

ivanetierra reviewed (approved):
new component is ok
new method to create challenge is ok (mock saved)
new rute is ok

ivilarop commented on src/app/modules/modals/login-modal/login-modal.component.spec.ts:
```diff
@@ -0,0 +1,101 @@
+  })
+
+  it('should call authService.login and success response if form is valid', async () => {
+    component.loginForm.setValue({ dni: '12345678Z', password: 'password' })
```
Sonar is failing. The problem identified is a hard-coded password in the code, which is a security risk. This happens because the password is written directly into the source code, which can expose credentials in public repositories or to insider attacks.you can consider using environment variables

ivilarop commented on src/app/modules/modals/login-modal/login-modal.component.spec.ts:
```diff
@@ -0,0 +1,101 @@
+  it('should handle login error', async () => {
+    const errorMessage: string = 'Usuario desactivado. Espere al email de confirmación'
+    const errorResponse = { status: 403 }
+    component.loginForm.setValue({ dni: '12345678Z', password: 'password' })
```
Sonar is failing. The problem identified is a hard-coded password in the code, which is a security risk. This happens because the password is written directly into the source code, which can expose credentials in public repositories or to insider attacks.you can consider using environment variables

ivilarop commented on src/app/modules/modals/register-modal/register-modal.component.spec.ts:
```diff
@@ -0,0 +1,128 @@
+      email: 'test@example.com',
+      name: 'testName',
+      itineraryId: 'testItinerary',
+      password: 'TestPassword1',
```
Sonar is failing. The problem identified is a hard-coded password in the code, which is a security risk. This happens because the password is written directly into the source code, which can expose credentials in public repositories or to insider attacks.you can consider using environment variables

ivilarop commented on src/app/modules/modals/register-modal/register-modal.component.spec.ts:
```diff
@@ -0,0 +1,128 @@
+      email: 'test@example.com',
+      name: 'testName',
+      itineraryId: 'testItinerary',
+      password: 'TestPassword1',
```
Sonar is failing. The problem identified is a hard-coded password in the code, which is a security risk. This happens because the password is written directly into the source code, which can expose credentials in public repositories or to insider attacks.you can consider using environment variables

ivilarop commented:
We should review the following bugs detected by sonar so that it passes the quality gate:
https://sonarcloud.io/project/issues?impactSoftwareQualities=RELIABILITY&severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&types=BUG&pullRequest=520&id=IT-Academy-BCN_ita-challenges-frontend

deuvede24 committed:
chore: update feature/509 branch with latest changes from develop

- Resolved conflicts and synchronized with develop branch
- Feature ready for challenge creation form implementation

deuvede24 committed:
chore: update dependencies after npm install

deuvede24 committed:
Merge remote-tracking branch 'origin/develop' into feature/509-create-basic-challenge-form

deuvede24 committed:
git commit -m "fix: Correct TypeScript version in package.json and package-lock.json to align with Angular 18 compatibility"
